core/src/main/java/oracle/weblogic/deploy/util/WLSDeployArchive.java

```diff
@@ -495,6 +495,52 @@ public boolean containsFileOrPath(String path) throws WLSDeployArchiveIOExceptio
         return result;
     }
 
+    /**
+     * Determines whether or not the path starts with the target pattern
+     * in the archive file.
+     *
+     * @param path the path into the archive file to test
+     * @param target the starts with target path
+     * @return true if the specified path starts with target
+     * @throws IllegalArgumentException    if the path is null or empty
+     */
+    public boolean isPathStartsWithTarget(String path, String target) throws IllegalArgumentException {
+        final String METHOD = "isApplicationsPath";
+
+        LOGGER.entering(CLASS, METHOD, path);
+        validateNonEmptyString(path, "path", METHOD);
+
+        boolean result = false;
+        // Verify that the path is into the binary root directory so that we do not allow random content.
+        if (path.startsWith(target)) {
+            result = true;
+        }
+        LOGGER.exiting(CLASS, METHOD, result);
+        return result;
+    }
+
+    /** Check whether the path should be used for remote update
+     *
+     * @param path
+     * @return true if the path should not be used for remote update
+     * @throws IllegalArgumentException    if the path is null or empty
+     */
+    public boolean isRemoteUpdateDomainForbiddenPath(String path)  throws IllegalArgumentException  {
```

Comment:
I'm not sure that this is the best place for this logic.  In my opinion, the archive file shouldn't know anything about remote mode.  Can't we put this into archive_helper.py?